openverse_catalog/dags/providers/provider_api_scripts/provider_data_ingester.py

```diff
@@ -114,12 +161,74 @@ def ingest_records(self, **kwargs) -> None:
                     logger.info("Batch complete.")
                     should_continue = False
 
-                if self.limit and record_count >= self.limit:
-                    logger.info(f"Ingestion limit of {self.limit} has been reached.")
-                    should_continue = False
-        finally:
-            total = self.commit_records()
-            logger.info(f"Committed {total} records")
+            except Exception as error:
```

Comment:
Based on the changes in #645, I think we'll need to be careful about bare `Exception` catches. It looks like if `skip_ingestion_errors` is `True`, then execution will continue. This breaks the case where the task is stopped by Airflow. I _think_ we can catch that specific case by handling `AirflowException` differently and immediately re-raising that though!

Reply:
Extremely good point. I think we'll also want to make sure we raise any ingestion errors that have previously been skipped before we encounter this.

What do you think about making `skip_ingestion_errors` a list of Exception types to skip, rather than a simple boolean?

Reply:
I like it! Only trouble would be matching on the exceptions since we have to go from a string to an exception type 😅 I guess we could just do `if exception.__type__.__name__ in skip_ingestion_errors`?